wiki/FitNesseRoot/HsacExamples/SlimTests/AndroidAppTests/content.txt

```diff
@@ -0,0 +1,46 @@
+﻿!-AndroidAppTest-! defines a language (a DSL of sorts) to interact with a native Android App. 
+ 
+Android App Test extends Browser Test for convenience methods, but needs an Appium !-androidDriver-! to interact with elements of type !-MobileElement-! (tap, swipe, etc.) This also means we can use Android's !-UiAutomator-! to locate interactable elements.
+
+locators are defined using !-UiSelectors-!. text is considered default, so looking for a given string will assume we are lokking for an element containing an exact match of this text.
+Otherwise, we can use all !-UiAutomator-! methods, by formatting our locator as method=value. Examples:
+
+-| comment |
+|Menu                           |Will match the first element that has a text value of 'Menu' (exact)                 |
+|text=Menu                      |Same as above                                                                        |
+|textMatches=/menu/i            |Will match the first element that matches the regex '/menu/i'                        |
+|textContains=Menu              |Will match the first element that has a text value containing 'Menu'                 |
+|textStartsWith=Menu            |Will match the first element that has a text value starting with 'Menu'              |
+|description=More options       |Will match the firste element that has the content-desc 'More options'(exact)        |
+|descriptionContains=options    |Will match the first element that has a content-desc containing 'options'            |
+|descriptionMatches=/OPT/i      |Will match the first element that has a content-desc that matches the regex '/OPT/i' |
+|descriptionStartsWith=More     |Will match the first element that has a content-desc starting with 'options'         |
+|resourceId=nl.pkg:id/error_text|Will match the element with resource id 'nl.pkg:id/error_text'(exact)                |
+|resourceIdMatches=/error_text/ |will match the first element with a resource id that matches the regex '/error_text/'|
+
+More info: [[Google documentation][http://developer.android.com/reference/android/support/test/uiautomator/UiSelector.html]]
+
+Example test cases:
+>ThuisBezorgdDemo (test Thuisbezorgd app installed on device)
+>NegenTweeAppDemo (install 9292 apk on device and run test)
+
+!2 Language
+!3 !-AndroidAppTest-!
+The main commands/keywords (i.e. public methods) offered by !-nl.hsac.fitnesse.fixture.slim.mobile.AndroidAppTest-! are listed below.
+The heuristic to find places tries a number of translations of end user terms to Selenium 'find' commands (e.g. is a full or partial label given, a link's or button's text, a placeholder). The first find that results in an element on the page is used by !-BrowserTest-!.
```

Comment:
Does the fixture have a heuristic (yet)? From the code and the text above I believe it either uses an exact match (default) or another locator. 

A heuristic, to me, would be something like first trying an exact match on text, then an exact match on accessibility identifier, and then partial matches on these.